File: SchedulingFront/src/app/common/models/menu.ts
import { Permission } from './permission';

export class Menu {
    menuId: number;
    code: string;
    name: string;
    image: string;
    url: string;
    sort: number;
    active: boolean;
    parentMenuId: number;
    clicked: boolean;
    children: Array<Menu> = new Array<Menu>();
    defaultPermissions: Array<Permission>;
    expanded: boolean;
    constructor(menu: Menu = null) {
        if (menu) {
            Object.assign(this, menu);
        }
        this.parentMenuId = null;
        this.expanded = false;
        this.clicked = false;
    }
}

File: SchedulingAPI/UserManagement/Models/Menu.cs
using Common.Extensions;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace UserManagement.Models
{
    public class Menu : ITreeview<Menu>
    {
        [PrimaryKey]
        public int MenuId { get; set; }
        public int? ParentId { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Url { get; set; }
        public string Image { get; set; }
        public int Sort { get; set; }
        public bool Active { get; set; }
        public int NotificationCount { get; set; } = 0;
        [ForeignKey("User")]
        public int UserId { get; set; }
        public List<Role> Role { get; set; }
        public List<Menu> Children { get; set; }

        public Menu()
        {
            Children = new List<Menu>();
            Role = new List<Role>();
        }
    }
}
